src/views/dog/MinigameWalk.vue

```diff
@@ -0,0 +1,332 @@
+const { goalPositionX, isGoalVisible, gameState, collectedPoos } = useGameState()
+const { characterAction, characterRef, triggerJump, handleJumpComplete } = useCharacterActions()
+const animatedElements = reactive([] as AnimatedComponentWithSpeedMultiplier[])
+const { spawnElementWithConfig, spawnInitialElements, lastElementSpawnTimes } =
+  useElementSpawning(animatedElements)
+
+const goalRef = ref<InstanceType<typeof Goal> | null>(null)
+const skylineRef = ref<HTMLDivElement | null>(null)
+const hasGameStarted = ref(false)
+let collisionTimeout: number | null = null
+let animationTimeline: gsap.core.Timeline | null = null
+const isAnimating = ref(false)
+
+const startGame = () => {
+  if (!hasGameStarted.value) {
+    run()
+    hasGameStarted.value = true
+  }
+}
+
+const handleButtonClick = () => {
+  if (hasGameStarted.value) {
+    triggerJump()
+  } else {
+    startGame()
+  }
+}
+
+const run = () => {
+  characterAction.value = 'run'
+  gsap.delayedCall(2, spawnObstacle)
+  animate()
+}
+
+const spawnPoo = () => {
+  characterAction.value = 'poop'
+  if (characterRef.value) {
+    const characterRect = characterRef.value.$el.getBoundingClientRect()
+    const positionX = characterRect.left + 30
+    gsap.delayedCall(0.9, () => {
+      gameState.poo = { positionX, collected: false }
+      gsap.delayedCall(0.4, () => {
+        characterAction.value = 'run'
+        animate()
+      })
+    })
+  }
+}
+
+const spawnObstacle = () => {
+  gameState.obstacle = { positionX: window.innerWidth }
+}
+
+const spawnElement = () => {
+  if (gameState.pooCount < 3 && Math.random() < 0.65) {
+    spawnPoo()
+  } else if (!isGoalVisible.value) {
+    spawnObstacle()
+  }
+}
+
+const collectPoo = () => {
+  if (gameState.poo) {
+    gameState.poo = null
+    gameState.pooCount++
+    checkGoalVisibility()
+    if (!isGoalVisible.value) {
+      spawnObstacle()
+    }
+  }
+}
+
+const handleJumpedOver = () => {
+  gameState.stonesJumped++
+  gameState.obstacle = null
+  checkGoalVisibility()
+  if (!isGoalVisible.value) {
+    spawnElement()
+  }
+}
+
+const checkGoalVisibility = () => {
+  if (gameState.pooCount >= 3 && gameState.stonesJumped >= 6) {
+    isGoalVisible.value = true
+    mascot.showMessage('STAGE3_SUPER')
+    goalPositionX.value = window.innerWidth + 100
+  }
+}
+
+const handleCollision = () => {
+  characterAction.value = 'hurt'
+  mascot.showMessage('STAGE3_OUTCH')
+  collisionTimeout = window.setTimeout(resetGame, 2000)
+}
+
+const animateElements = () => {
+  const duration = 0.01
+  const distance = 4
+  animatedElements.forEach((element, index) => {
+    animationTimeline?.to(
+      element,
+      { positionX: `-=${distance * element.speedMultiplier}`, duration },
+      0
+    )
+    if (element.positionX < -500) {
+      animatedElements.splice(index, 1)
+    }
+  })
+
+  if (!skylineRef.value) return
+  animationTimeline?.to(
+    skylineRef.value,
+    {
+      backgroundPositionX: `-=${distance * 0.3}`,
```

Comment:
With a parallax effect, the elements in the foreground should move faster than those in the background (distance * 0.05 should work)